src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp

```diff
@@ -107,6 +111,16 @@ void QgsExternalResourceConfigDlg::rootPathPropertyChanged()
   mRootPathButton->setEnabled( !prop.isActive() );
 }
 
+void QgsExternalResourceConfigDlg::documentViewerContentPropertyChanged()
+{
+  QgsProperty prop = mDocumentViewerContentPropertyOverrideButton->toProperty();
+  setDocumentViewerContentExpression( prop );
+
+  mDocumentViewerContentExpression->setVisible( prop.isActive() );
+  //mDocumentViewerContent->setVisible( !prop.isActive() );
+  mDocumentViewerContentComboBox->setEnabled( !prop.isActive() );
```

Comment:
fixed. I added for visible and enable states (with option to have it the other way around)